src/main/java/org/osc/controller/nsc/restserver/api/InspectionApis.java

```diff
@@ -0,0 +1,403 @@
+        props.setProperty(JDBC_USER, "admin");
+        props.setProperty(JDBC_PASSWORD, "admin123");
+        DataSource ds = null;
+        try {
+            ds = this.jdbcFactory.createDataSource(props);
+        } catch (SQLException error) {
+            LOG.error(error);
+            throw new IllegalStateException(error.getMessage(), error);
+        }
+        EntityManager em = this.resourceFactory
+                .getProviderFor(this.builder, singletonMap("javax.persistence.nonJtaDataSource", (Object) ds), null)
+                .getResource(this.txControl);
+        this.em = em;
+        this.api = new SampleSdnRedirectionApi(this.txControl, this.em);
+        this.utils = new RedirectionApiUtils(this.em, this.txControl);
+        this.ipAddr = ipAddr;
+
+    }
+
+    @Path("/ipAddr/{ipAddr}/netelem")
+    @POST
+    public String createNetworkElement(@PathParam("ipAddr") String ipAddr, NetworkElementEntity entity)
+            throws Exception {
+        if (entity == null) {
+            throw new IllegalArgumentException("Attempt to create network element with no network entities");
+        }
+        LOG.info(String.format("Creating network element with elementId  id %s)", "" + entity.getElementId()));
+
+        init(ipAddr);
+        return this.txControl.required(() -> {
+            NetworkElementEntity element = this.utils.txNetworkElementEntityByElementId(entity.getElementId());
+
+            if (element != null) {
+                String msg = String.format("Network element already exists id: %s\n", entity.getElementId());
+                LOG.error(msg);
+                throw new IllegalArgumentException(msg);
+            }
+
+            this.em.merge(entity);
+            return entity.getElementId();
+        });
+    }
+
+    @Path("/ipAddr/{ipAddr}/netelem/{elementId}")
+    @PUT
+    public NetworkElementEntity updateNetworkElement(@PathParam("ipAddr") String ipAddr,
+            @PathParam("elementId") String elementId, NetworkElementEntity entity) throws Exception {
+        LOG.info(String.format("Updating the network element id %s ", "" + elementId));
+        init(ipAddr);
+
+        return this.txControl.required(() -> {
+            NetworkElementEntity element = this.utils.txNetworkElementEntityByElementId(entity.getElementId());
+
+            if (element == null) {
+                String msg = String.format("Cannot find the network element id: %s\n", entity.getElementId());
+                LOG.error(msg);
+                throw new IllegalArgumentException(msg);
+            }
+
+            this.em.merge(entity);
+            return entity;
+        });
+    }
+
+    @Path("/ipAddr/{ipAddr}/netelem/{elementId}")
+    @DELETE
+    public void deleteNetworkElement(@PathParam("ipAddr") String ipAddr, @PathParam("elementId") String elementId)
+            throws Exception {
+        LOG.info(String.format("Deleting the network element for id %s ", "" + elementId));
```

Comment:
"" not needed. Also look at other places

Reply:
ok